editor/code_editor.cpp

```diff
@@ -607,7 +607,7 @@ void FindReplaceBar::_search_text_submitted(const String &p_text) {
 }
 
 void FindReplaceBar::_replace_text_submitted(const String &p_text) {
-	if (selection_only->is_pressed() && text_editor->has_selection(0)) {
+	if ((Input::get_singleton()->is_key_pressed(Key::CMD_OR_CTRL) && Input::get_singleton()->is_key_pressed(Key::ALT)) || (selection_only->is_pressed() && text_editor->has_selection(0))) {
```

Comment:
The feature makes sense, but this isn't the right way to add shortcuts.

BaseButton has an API for this already, which should be usable here, a priori (didn't test).
This lets you register a shortcut, with a property name so it can be customized by the user, and it gets added to a tooltip automatically with the proper localization. If the shortcut is pressed, the button is pressed and triggers its callback.

In the constructor:
```cpp
replace_all->set_shortcut(ED_SHORTCUT("script_editor/replace_all_occurrences", TTR("Replace All Occurrences"), KeyModifierMask::CMD_OR_CTRL | KeyModifierMask::ALT | Key::ENTER);
```

Reply:
Thanks for this! As a button on its own with a shortcut, I'd do this with no hesitation, but I do have to raise the question of what to do with the rest of the interactions that don't have associated buttons?

e.g. `Shift-Enter` to search / replace backwards?

And also, the plain `Enter` isn't a mapping for a search / replace shortcut either - it's just a submission from a `TextEdit`.

Is this asymmetry okay?

Reply:
The actions which don't have an associated button should rely on `ui_` input actions in the InputMap, and possibly add new ones if they're not covered already.

But there are indeed still a bunch of places where we hardcode specific keys instead of relying on action, I don't know if it's an oversight (UI actions for stuff like TextEdit were added recently), or if there's a good reason for it.

CC @KoBeWi

Reply:
More like CC @Paulb23

Reply:
For the time being, should I add this shortcut as described by @akien-mga, without any further modifications to the other interactions?

Reply:
Yes I would say so.

Reply:
Unfortunately the shortcut only works when none of the text fields have focus. When one of them has focus, the text submission shortcut takes priority. Any idea how to get around that?

Reply:
I think this solution is okay as it's only checking modifiers, from the `LineEdit`'s `text_submitted` signal. Which should already be using the `ui_text_submit` shortcut.

Given it's not currently handled like a shortcut / input event as above, we'd have to do some restructuring to support this like we do in `CodeTextEditor::input`, and bypass the `LineEdit` signal.

Reply:
So for this PR, what's the best course of action? Leave it as a modifier check? Or restructuring?